src/Components/Server/src/Prerendering/ComponentPrerenderResult.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Components.Server.Prerendering
+{
+    /// <summary>
+    /// Represents the result of a prerendering an <see cref="IComponent"/>.
+    /// </summary>
+    public sealed class ComponentPrerenderResult
```

Comment:
How is this different from the other type that's returned from rendering?

Reply:
I plan to change ComponentRenderedText in the future to support rendering without buffering the contents into intermediate strings. As part of that, the API will evolve to have a WriteTo(HtmlEncoder encoder, TextWriter output) and won't have any public ctor (as its going to internally use the render tree frames).

I didn't do that as part of this change because it would add unnecessary complexity to this changeset, but once I do that, things outside the components assembly won't be able to "augment" that struct.

You can think of ComponentPrerenderResult as the general type used for prerendering and of ComponentRenderedText as the high performance lowlevel type responsible for serializing the component output.